python/lsst/meas/deblender/oldBaseline.py

```diff
@@ -0,0 +1,1420 @@
+
+        return heavy
+
+    def setStrayFlux(self, stray):
+        self.strayFlux = stray
+
+    def setFluxPortion(self, mimg):
+        self.fluxPortion = mimg
+
+    def setTemplateWeight(self, w):
+        self.templateWeight = w
+
+    def setPatched(self):
+        self.patched = True
+
+    # DEBUG
+    def setOrigTemplate(self, t, tfoot):
+        self.origTemplate = t.Factory(t, True)
+        self.origFootprint = tfoot
+
+    def setRampedTemplate(self, t, tfoot):
+        self.hasRampedTemplate = True
+        self.rampedTemplate = t.Factory(t, True)
+
+    def setMedianFilteredTemplate(self, t, tfoot):
+        self.medianFilteredTemplate = t.Factory(t, True)
+
+    def setPsfTemplate(self, tim, tfoot):
+        self.psfFootprint = afwDet.Footprint(tfoot)
+        self.psfTemplate = tim.Factory(tim, True)
+
+    def setOutOfBounds(self):
+        self.outOfBounds = True
+        self.skip = True
+
+    def setTinyFootprint(self):
+        self.tinyFootprint = True
+        self.skip = True
+
+    def setNoValidPixels(self):
+        self.noValidPixels = True
+        self.skip = True
+
+    def setPsfFitFailed(self):
+        self.psfFitFailed = True
+
+    def setBadPsfDof(self):
+        self.psfFitBadDof = True
+
+    def setDeblendedAsPsf(self):
+        self.deblendedAsPsf = True
+
+    def setFailedSymmetricTemplate(self):
+        self.failedSymmetricTemplate = True
+        self.skip = True
+
+    def setTemplate(self, image, footprint):
+        self.templateImage = image
+        self.templateFootprint = footprint
+
+
+def deblend(footprint, maskedImage, psf, psffwhm,
+            psfChisqCut1=1.5, psfChisqCut2=1.5, psfChisqCut2b=1.5, fitPsfs=True,
+            medianSmoothTemplate=True, medianFilterHalfsize=2,
+            monotonicTemplate=True, weightTemplates=False,
+            log=None, verbose=False, sigma1=None, maxNumberOfPeaks=0,
+            findStrayFlux=True, assignStrayFlux=True,
+            strayFluxToPointSources='necessary', strayFluxAssignment='r-to-peak',
+            rampFluxAtEdge=False, patchEdges=False, tinyFootprintSize=2,
+            getTemplateSum=False,
+            clipStrayFluxFraction=0.001, clipFootprintToNonzero=True,
+            removeDegenerateTemplates=False, maxTempDotProd=0.5
+            ):
+    """Deblend a parent ``Footprint`` in a ``MaskedImageF``.
+    
+    Deblending assumes that ``footprint`` has multiple peaks, as it will still create a
+    `PerFootprint` object with a list of peaks even if there is only one peak in the list.
+    It is recommended to first check that ``footprint`` has more than one peak, similar to the
+    execution of `lsst.meas.deblender.deblend.SourceDeblendTask`.
+    
+    Deblending involves several mandatory and optional steps:
+    # Optional: If ``fitPsfs`` is True, find all peaks that are well-fit by a PSF + background model
+        * Peaks that pass the cuts have their footprints modified to the PSF + background model
```

Comment:
I'm going to ignore the comments in oldBaseline.py, since this is just an exact copy of the current `baseline.py` file from master, which I can delete before merging this ticket. I just saved it to be able to run tests to make sure that the new API it bitwise the same as the old one.